picoquic/frames.c

```diff
@@ -1795,11 +1795,24 @@ int picoquic_decode_max_stream_id_frame(picoquic_cnx_t* cnx, uint8_t* bytes,
  * Sending of miscellaneous frames
  */
 
-int picoquic_prepare_misc_frame(picoquic_cnx_t* cnx, uint8_t* bytes,
-    size_t bytes_max, size_t* consumed)
+int picoquic_prepare_first_misc_frame(picoquic_cnx_t* cnx, uint8_t* bytes,
+                                      size_t bytes_max, size_t* consumed)
+{
+    int ret = picoquic_prepare_misc_frame(cnx->first_misc_frame, bytes, bytes_max, consumed);
+
+    if (ret == 0) {
+        picoquic_misc_frame_header_t* misc_frame = cnx->first_misc_frame;
+        cnx->first_misc_frame = misc_frame->next_misc_frame;
+        free(misc_frame);
+    }
```

Comment:
Maybe we need to start write documentation for these API. Starting on the Wiki would be nice. I am not sure that I understand the difference between picoquic_prepare_first_misc_frame() and picoquic_prepare_misc_frame(). The original function was supposed to take the first frame in the queue, copy it to the output buffer, and then remove it from the queue and delete the corresponding buffer.

This matches the behavior of the posting function, which takes the input from the application, allocates a buffer, copies the frame in the buffer, and chains the buffer to the misc queue.